### PR TITLE
must-gather: collects output for storageclassclaim

### PR DESCRIPTION
**Pull request by yati1998**
This commit collects output for get and describe
of storageclassclaim for all namespaces and also
collects the yaml file for it.

Signed-off-by: yati1998 <ypadia@redhat.com>

### CODE REVIEW

**Comment by yati1998** [comment]
@Madhu-1 would like to get your review here.

File: must-gather/collection-scripts/gather_namespaced_resources [hunk 1]
```diff
@@ -156,3 +156,16 @@ echo "collecting dump of oc get volumereplication all namespaces" | tee -a  "${B
 echo "collecting dump of oc get volumereplicationgroups all namespaces" | tee -a  "${BASE_COLLECTION_PATH}/gather-debug.log"
 { oc get volumereplicationgroups --all-namespaces; } >> "${BASE_COLLECTION_PATH}/namespaces/all/vrg_all_namespaces"
 { oc adm --dest-dir="${BASE_COLLECTION_PATH}/namespaces/all/" inspect vrg --all-namespaces --"${SINCE_TIME}"; } >> "${BASE_COLLECTION_PATH}/gather-debug.log" 2>&1
+
+# Collect details of storageclassclaim for managed services
+ echo "collecting dump of oc get storageclassclaim" | tee -a  "${BASE_COLLECTION_PATH}/gather-debug.log"
+namespaces=$(oc get storageclassclaim -A --no-headers | awk '{print $1}')
```
Comment (Madhu-1) [inline must-gather/collection-scripts/gather_namespaced_resources:162]: why not `oc get namespaces` here?

**Comment by Madhu-1** [comment]
cc @nb-ohad

File: must-gather/collection-scripts/gather_namespaced_resources (hunk 1 above)
Comment (yati1998) [reply must-gather/collection-scripts/gather_namespaced_resources:162]: `oc get namespaces` shall list all the namespaces right? How do we get under which ns storageclassclaim is created? Am I missing anything?

File: must-gather/collection-scripts/gather_namespaced_resources [hunk 2]
```diff
@@ -156,3 +156,16 @@ echo "collecting dump of oc get volumereplication all namespaces" | tee -a  "${B
...
+for ns in "${namespaces}"; do
```
Comment (jilju) [inline must-gather/collection-scripts/gather_namespaced_resources:163]: If there are more than one storageclassclaim in a particular namespace, the output file will be appended with duplicate content.

File: must-gather/collection-scripts/gather_namespaced_resources (hunk 1 above)
Comment (Madhu-1) [reply must-gather/collection-scripts/gather_namespaced_resources:162]: got it. it should be fine. this avoids listing all namespaces which doesn't have storageclassclaim

File: must-gather/collection-scripts/gather_namespaced_resources (hunk 2 above)
Comment (yati1998) [reply must-gather/collection-scripts/gather_namespaced_resources:163]: Thanks for notifying @jilju, will update

Comment (Madhu-1) [reply must-gather/collection-scripts/gather_namespaced_resources:163]: you need to remove duplicate namespace entries. or use same way we do for other resources like VR,VSC etc

File: must-gather/collection-scripts/gather_namespaced_resources (hunk 1 above)
Comment (yati1998) [reply must-gather/collection-scripts/gather_namespaced_resources:162]: yes.

File: must-gather/collection-scripts/gather_namespaced_resources (hunk 2 above)
Comment (yati1998) [reply must-gather/collection-scripts/gather_namespaced_resources:163]: yes probably, checking that

File: must-gather/collection-scripts/gather_namespaced_resources [hunk 3]
```diff
@@ -156,3 +156,9 @@ echo "collecting dump of oc get volumereplication all namespaces" | tee -a  "${B
 echo "collecting dump of oc get volumereplicationgroups all namespaces" | tee -a  "${BASE_COLLECTION_PATH}/gather-debug.log"
 { oc get volumereplicationgroups --all-namespaces; } >> "${BASE_COLLECTION_PATH}/namespaces/all/vrg_all_namespaces"
 { oc adm --dest-dir="${BASE_COLLECTION_PATH}/namespaces/all/" inspect vrg --all-namespaces --"${SINCE_TIME}"; } >> "${BASE_COLLECTION_PATH}/gather-debug.log" 2>&1
+
+# Collect details of storageclassclaim of all namespaces for managed services
+echo "collecting dump of oc get storageclassclaim all namespaces" | tee -a  "${BASE_COLLECTION_PATH}/gather-debug.log"
+{ oc get storageclassclaim --all-namespaces; } >> "${BASE_COLLECTION_PATH}/namespaces/all/storageclassclaim_all_ns"
```
Comment (Madhu-1) [inline must-gather/collection-scripts/gather_namespaced_resources:162]: instead of `storageclassclaim_all_ns` use different files for each command?

Comment (yati1998) [reply must-gather/collection-scripts/gather_namespaced_resources:162]: for that, we need to specify individual namespaces and collect them, as it was done in the previous commit, and for that, we need to get unique namespaces.  I would prefer this way as other resources are also collected in a similar way. Still, let me know if there is any specific reason to collect it in different files?

Comment (Madhu-1) [reply must-gather/collection-scripts/gather_namespaced_resources:162]: i meant like below

```bash
echo "collecting dump of oc get storageclassclaim all namespaces" | tee -a  "${BASE_COLLECTION_PATH}/gather
debug.log"
{ oc get storageclassclaim --all-namespaces; } >> "${BASE_COLLECTION_PATH}/namespaces/all/get_storageclassclaim_all_ns"
{ oc describe storageclassclaim --all-namespaces; } >> "${BASE_COLLECTION_PATH}/namespaces/all/desc_storageclassclaim_all_ns"
{ oc get storageclassclaim -oyaml --all-namespaces; } >> "${BASE_COLLECTION_PATH}/namespaces/all/get_yaml_storageclassclaim_all_ns"
```

Comment (yati1998) [reply must-gather/collection-scripts/gather_namespaced_resources:162]: oh, okay that. Yeah, that will be fine. Will update it.

**Review by Madhu-1 — APPROVED** [review]

**Review by agarwal-mudit — COMMENTED** [review]
What happens in case of non managed services clusters?

**Comment by yati1998** [comment]
> What happens in case of non managed services cluster

> What happens in case of non managed services clusters?

Yeah seems we need to add a check to make sure storageclassclaim exists!!

**Comment by yati1998** [comment]
@agarwal-mudit this is ready for review now.

**Review by Madhu-1 — APPROVED** [review]
@yati1998 Have you tested this change in any MS cluster?

**Review by agarwal-mudit — APPROVED** [review]

**Pull request merged**